### PR TITLE
Updates for updated splinepy

### PR DESCRIPTION
**Pull request by j042**
# Overview
splinepy had some changes and gustaf requires some changes before we setup a CD.
@clemens-fricke: could you please check FFD compatibility / make updates?
@jzwar: microstructure!
We can get rid of a lot of type conversions and safety copying

### CODE REVIEW

**Comment by jzwar** [comment]
Is there now a more generic way to create an interface to the inherited splines in gustaf. I am thinking primarily of the load functions as well as split functions where new splines are generated...

**Comment by j042** [comment]
> Is there now a more generic way to create an interface to the inherited splines in gustaf. I am thinking primarily of the load functions as well as split functions where new splines are generated...

maybe something like 
`gustaf.spline.settings.NAME_TO_TYPE["<spline-name>"](**dictspline)`
or
`gustaf.spline.settings.NAME_TO_TYPE["<spline-name>"](spline=already_initialized_PySpline)`

**Comment by j042** [comment]
ufff

File: gustaf/helpers/raise_if.py [hunk 1]
```diff
@@ -47,16 +47,15 @@ class ModuleImportRaiser():
...
-        "Parts of the requested functionality in gustaf depend on the "
-        f"external `{lib_name}` package which could not be found on "
-        "your system. Please refer to the installation instructions "
-        "for more information. "
-        f"{f'Original error message {error_message}' if error_message else ''}"
+                "Parts of the requested functionality in gustaf depend on the "
+                f"external `{lib_name}` package which could not be found on "
+                "your system. Please refer to the installation instructions "
+                "for more information. "
+                f"Original error message - {error_message}"
```
Comment (clemens-fricke) [inline gustaf/helpers/raise_if.py:56]: This will now always print the Original error message part even if none is given.

Comment (j042) [reply gustaf/helpers/raise_if.py]: what about now?

Comment (j042) [reply gustaf/helpers/raise_if.py]: we can probably replace documentation address with `pip` or `conda` commands

**Review by clemens-fricke — APPROVED** [review]
YAS, wait a second for better piccolo window size adaptation

**Review by clemens-fricke — APPROVED** [review]

**Pull request merged**